pkg/kubelet/volumemanager/populator/desired_state_of_world_populator.go

```diff
@@ -278,12 +278,12 @@ func (dswp *desiredStateOfWorldPopulator) findAndRemoveDeletedPods() {
 			klog.V(4).InfoS("Pod still has one or more containers in the non-exited state and will not be removed from desired state", "pod", klog.KObj(volumeToMount.Pod))
 			continue
 		}
-		exists, _, _ := dswp.actualStateOfWorld.PodExistsInVolume(volumeToMount.PodName, volumeToMount.VolumeName)
+		removed := dswp.actualStateOfWorld.PodRemovedFromVolume(volumeToMount.PodName, volumeToMount.VolumeName)
 		var volumeToMountSpecName string
```

Comment:
so if volume was uncertain, this would result in volume being deleted from DSOW. Previously we did exact opposite:

> if volume mount was uncertain we should keep trying to mount the volume

which means if volume-mgr started mounting the volume and say first mount failed and volume entered uncertain state, then we do not remove the volume from DSOW until volume is correctly mounted.  Now we would basically abort potentially in-flight mount request and issue umount requests.  I am not sure which one is more correct.

Reply:
I think it is fine.

Reply:
> if volume mount was uncertain we should keep trying to mount the volume

We want to keep this behavior when a pod needs the volume. However, the change above is for deleted pods, where we want to "keep trying to **unmount** the volume". The mount may never succeed, because of timeouts or whatnot.